File: MD.CMS.Administration/MD.CMS.Administration.Core.Web/scripts/businessLogic/dataAccess/entities/profileTypeFieldValue.ts
/// <reference path="./base/iBaseEntity.ts" />
/// <reference path="./profileTypeField.ts" />
/// <reference path="./genericContent/genericContentFieldValue.ts" />

namespace mdBusinessLogic.dataAccess.entities {
    export class profileTypeFieldValue extends entities.genericContent.genericContentFieldValue implements base.IBaseEntity<profileTypeFieldValue> {
        public ProfileTypeFieldId: number;
        public ProfileTypeId: number;
        public UserId: number;

        constructor(obj?: profileTypeFieldValue) {
            super(obj);
            this.ProfileTypeFieldId = 0;
            this.ProfileTypeId = 0;
            this.UserId = 0;
            if (obj != undefined && obj != null) {
                this.construct(obj);
            }
        }

        construct(data: any): void {
            super.construct(data);
            this.ProfileTypeFieldId = this.getValue<number>(data, "ProfileTypeFieldId", 0);
            this.ProfileTypeId = this.getValue<number>(data, "ProfileTypeId", 0);
            this.UserId = this.getValue<number>(data, "UserId", 0);
        }

        public clone(): profileTypeFieldValue {
            return new profileTypeFieldValue(this);
        }

    }
}

File: MD.CMS.BusinessLogic.Core/DataAccess/Entities/ProfileTypeFieldValue.cs
using MD.CMS.BusinessLogic.Core.DataAccess.Entities.GenericContent;

namespace MD.CMS.BusinessLogic.Core.DataAccess.Entities
{
    public class ProfileTypeFieldValue : GenericContentFieldValue
    {
        #region Attributes
        private long _profileTypeFieldId;
        private long _profileTypeId;
        private string _userId;
        #endregion

        #region Properties
        public long ProfileTypeFieldId
        {
            get { return _profileTypeFieldId; }
            set { _profileTypeFieldId = value; }
        }

        public long ProfileTypeId
        {
            get { return _profileTypeId; }
            set { _profileTypeId = value; }
        }

        public string UserId
        {
            get { return _userId; }
            set { _userId = value; }
        }
        #endregion


        public ProfileTypeFieldValue()
        {

        }

        public ProfileTypeFieldValue(ProfileTypeField obj) : base(obj)
        {
            if (obj != null)
            {
                ProfileTypeFieldId = obj.Id;
                ProfileTypeId = obj.ProfileTypeId;
            }
        }

        public ProfileTypeFieldValue(ProfileTypeFieldValue obj) : base(obj)
        {
            if (obj != null)
            {
                ProfileTypeFieldId = obj.Id;
                ProfileTypeId = obj.ProfileTypeId;
                UserId = obj.UserId;
            }
        }

        public ProfileTypeField ToProfileTypeField()
        {
            return new ProfileTypeField(this);
        }
    }
}
